storm-core/src/clj/org/apache/storm/daemon/common.clj

```diff
@@ -298,18 +299,21 @@
                       {[comp-id METRICS-STREAM-ID] :shuffle})
                     (into {}))
         
-        mk-bolt-spec (fn [class arg p]
+        mk-bolt-spec (fn [class arg p max-retain-metric-tuples whitelist blacklist]
                        (thrift/mk-bolt-spec*
                         inputs
-                        (org.apache.storm.metric.MetricsConsumerBolt. class arg)
+                        (org.apache.storm.metric.MetricsConsumerBolt. class arg max-retain-metric-tuples (FilterByMetricName. whitelist blacklist))
```

Comment:
@ptgoetz 
I intended to make it pluggable, but if we just want to allow filtering by name, pluggable incurs unneeded specifying class name. So flexible vs easy to use.
I'd like to get opinions on this side, and happy to change when we want to make it flexible.